libr/bin/p/bin_xtr_fatmach0.c

```diff
@@ -92,7 +92,14 @@ static RBinXtrData * extract(RBin* bin, int idx) {
 	arch = r_bin_fatmach0_extract (fb, idx, &narch);
 	if (!arch) return res;
 
-	res = r_bin_xtrdata_new (NULL, NULL, arch->b, arch->offset, arch->size, narch);
+	RBinXtrMetadata *metadata = R_NEW0(RBinXtrMetadata);
+
+	metadata->bits = MACH0_(get_bits)(r_buf_buffer(arch->b));
```

Comment:
Still i am not very keen with this solution. I would rather embed the logic of get bits and arch inside dyldcache.c

```C

int is_magic_64(ut32 magic) {
  return magic == MH_MAGIC_64 || magic == MH_CIGAM_64;
}
struct MACH0_(mach_header) hdr;
magic = (ut32)buf;
if (magic == MACH0_(MH_MAGIC))
     big_endian = false;
else if (magic == MACH0_(MH_CIGAM))
     big_endian = true;
else if (magic == FAT_CIGAM)
     big_endian = true;
else 
     return false; 
len = r_buf_fread_at (bin->b, 0, (ut8*)&hdr, is_magic_64? (big_endian?"8I":"8i"):(big_endian?"7I":"7i"), 1);

```

Then get arch and bits from hdr